src/boot/efi/part-discovery.c

```diff
@@ -168,11 +168,6 @@ static EFI_STATUS try_gpt(
                 };
                 memcpy(ret_hd->Signature, &entry->UniquePartitionGUID, sizeof(ret_hd->Signature));
 
-                /* HARDDRIVE_DEVICE_PATH has padding, which at least OVMF does not like. */
-                SetDevicePathNodeLength(
-                                &ret_hd->Header,
-                                offsetof(HARDDRIVE_DEVICE_PATH, SignatureType) + sizeof(ret_hd->SignatureType));
```

Comment:
Why dropped? Is this change relevant to this commit?

Reply:
The removed comment hints at it: our definition of the struct doesn't have the padding, so we can use sizeof directly when initializing it a few lines above.

Reply:
Thanks.